.changeset/red-rings-tell.md

```diff
@@ -0,0 +1,38 @@
+---
+'astro': patch
+---
+
+Adds additional validation options to `astro:env`
+
+`astro:env` schema datatypes `string` and `number` now have new optional validation rules:
+
+```js
+import { defineConfig, envField } from 'astro/config'
+
+export default defineConfig({
+    experimental: {
+        env: {
+            schema: {
+                FOO: envField.string({
+                    // ...
+                    max: 32,
+                    min: 3,
+                    length: 12,
+                    url: true,
```

Comment:
What does this do, does it turn it into a `URL` object? If so I feel like a new type of URL, `envField.url()` might be better.

Reply:
It's the equivalent of `z.string().url()`, so it only returns a string that is a valid url. Happy to add `envField.url` if some users request it